### PR TITLE
Rework examples to use csolution layers

### PR DESCRIPTION
**Pull request by VladimirUmek**

### CODE REVIEW

**Comment by JonatanAntoni** [comment]
Shall we add automated testing for the examples?

**Comment by VladimirUmek** [comment]
That would be highly beneficial if it can be easily done.

**Review by JonatanAntoni — APPROVED** [review]

File: CMSIS/RTOS2/FreeRTOS/Examples/Examples.csolution.yml [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+  packs:
+    - pack: ARM::CMSIS@>=6.0.0-0
+    - pack: ARM::Cortex_DFP
+
+  target-types:
+    - type: CM0
+      device: ARMCM0
+      variables:
+        - Target-Layer: ../../Target/CM0/Target.clayer.yml
```
Comment (JonatanAntoni) [inline CMSIS/RTOS2/FreeRTOS/Examples/Examples.csolution.yml:14]: These relative paths look weird. Shouldn't they be relartive to the csolution file?

Comment (VladimirUmek) [reply CMSIS/RTOS2/FreeRTOS/Examples/Examples.csolution.yml:14]: I don't know of any alternative way of specifying this given the solution structure.

Comment (JonatanAntoni) [reply CMSIS/RTOS2/FreeRTOS/Examples/Examples.csolution.yml:14]: Sorry, I was mislead.

**Pull request merged**